dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -1,22 +1,92 @@
 from pathlib import Path
 
+from dpytools.stores.directory.local import LocalDirectoryStore
+from dpypelines.pipeline.shared import notification as notify
+from dpypelines.pipeline.shared.pipelineconfig import matching
+from dpypelines.pipeline.functions.config import get_pipeline_identifier_from_config
+from pipelines.pipeline.shared import message
 
-# IMPORTANT
-# We probably want to break these stages out a bit
-# to help with reusibility
-def dataset_ingress_v1(files_dir: Path):
+def dataset_ingress_v1(files_dir: str) -> None:
     """
     Version one of the dataset ingress pipeline.
 
     files_dir: Path to the directory where the input
     files for this pipeline are located.
+
+    Args:
+        files_dir (str): Path to the directory where the input files for this pipeline are located.
+
+    Raises:
+        ValueError: If required files, supplementary distributions, or pipeline configuration are not found in the input directory.
+        Exception: If any other unexpected error occurs.
     """
 
-    # verify that the specified required files have been provided
+    # Attempt to access the local data store
+    try:
+        local_store = LocalDirectoryStore(files_dir)
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to access local data at {files_dir}", err))
+        raise err
+
+    # Load the pipeline configuration as a dictionary
+    try:
+        pipeline_config: dict = local_store.get_lone_matching_json_as_dict("pipeline-config.json")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error("Failed to get pipeline-config.json", err))
+        raise err
+
+    
+    # Make sure pipeline_config contains a "pipeline" key
+    try:
+        pipeline = get_pipeline_identifier_from_config(pipeline_config)
+    except KeyError:
+        notify.data_engineering(message.expected_config_key_missing("Pipeline key not found in config", "pipeline", "dataset_ingress_v1"))
```

Comment:
With this (and a few other blocks) we're generating the message twice, once for the notify and once for the exception. It's a bit messy and opens the door to someone desycning them by changing one and not the other.

just assign it, i.e `msg = message.expected_config_key_missing("Pipeline key not found in config", "pipeline", "dataset_ingress_v1")` and utilise that variable in both places.

I wont comment them all but there's a few you can switch over.